src/Voron/Impl/Paging/CryptoPager.cs

```diff
@@ -12,9 +14,47 @@
 
 namespace Voron.Impl.Paging
 {
-    public class CryptoTransactionState
+    public class CryptoTransactionState: IEnumerable<KeyValuePair<long, EncryptionBuffer>>
     {
-        public Dictionary<long, EncryptionBuffer> LoadedBuffers = new Dictionary<long, EncryptionBuffer>();
+        private Dictionary<long, EncryptionBuffer> _loadedBuffers = new Dictionary<long, EncryptionBuffer>();
+        private long _totalCryptoBufferSize;
+
+        /// <summary>
+        /// Used for computing the total memory used by the transaction crypto buffers
+        /// </summary>
+        public long TotalCryptoBufferSize => _totalCryptoBufferSize;
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool TryGetValue(long i, out EncryptionBuffer value)
+        {
+            return _loadedBuffers.TryGetValue(i, out value);
+        }
+
+        public EncryptionBuffer this[long index]
+        {
+            get => _loadedBuffers[index];
+            //This assumes that we don't replace buffers just set them.
+            set
+            {
+                _loadedBuffers[index] = value;
+                _totalCryptoBufferSize += value.Size;
+            }
+        }
+
+        public IEnumerator<KeyValuePair<long, EncryptionBuffer>> GetEnumerator()
+        {
+            return _loadedBuffers.GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public void ReverseBuffers()
```

Comment:
What is this? 
This has no meaning, dictionary has no defined order.

Reply:
i'm the wrong guy to complain about this:
https://github.com/ravendb/ravendb/blame/v4.2/test/SlowTests/Voron/Issues/RavenDB_10825.cs#L40